### PR TITLE
CE#34567: Fix deprecation issues found by integration tests

### PR DESCRIPTION
**Pull request by karyna-t**
### Description (*)
Fix deprecation warning found by Integration Tests for PHP 8.1. compatibility

### Related Issues
1. magento/magento2#34567

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by andrewbess** [comment]
@magento run Static Tests

File: app/code/Magento/Customer/Block/Adminhtml/Edit/Tab/View/PersonalInfo.php [hunk 1]
```diff
@@ -439,7 +417,8 @@ public function getLastLoginDate()
      */
     public function getStoreLastLoginDate()
     {
-        $date = strtotime($this->getCustomerLog()->getLastLoginAt());
+        $lastLogin = $this->getCustomerLog()->getLastLoginAt();
```
Comment (karyna-t) [inline app/code/Magento/Customer/Block/Adminhtml/Edit/Tab/View/PersonalInfo.php:420]: app/code/Magento/Customer/Model/Logger.php:82

**Comment by andrewbess** [comment]
@magento run all tests

**Comment by andrewbess** [comment]
@magento run Integration Tests, Functional Tests CE, Functional Tests EE, Functional Tests B2B

File: app/code/Magento/Directory/Model/Currency.php [hunk 2]
```diff
@@ -488,7 +488,7 @@ private function setOptions(array $options): void
             $this->numberFormatter->setSymbol(\NumberFormatter::CURRENCY_SYMBOL, '');
         }
         if (array_key_exists('precision', $options)) {
-            $this->numberFormatter->setAttribute(\NumberFormatter::FRACTION_DIGITS, $options['precision']);
+            $this->numberFormatter->setAttribute(\NumberFormatter::FRACTION_DIGITS, $options['precision'] ?? 0);
```
Comment (xmav) [inline app/code/Magento/Directory/Model/Currency.php]: I don't think thats right place for fix. There are few places that use `null` as a value for precision while default is 2 \Magento\Framework\Pricing\PriceCurrencyInterface::DEFAULT_PRECISION
https://github.com/magento/magento2/blob/548ac72cab788d7ad09393cb838cce39d399061d/lib/internal/Magento/Framework/Pricing/PriceCurrencyInterface.php#L50-L56

I would ask you to test and see if we really expect\want `0` as precision in 
https://github.com/magento/partners-magento2b2b/blob/32b75f9bc94fb4ddf9aa3a07393179554406eab8/app/code/Magento/PurchaseOrder/Ui/Component/Listing/Column/Price.php#L81-L86

https://github.com/magento/partners-magento2b2b/blob/32b75f9bc94fb4ddf9aa3a07393179554406eab8/app/code/Magento/CompanyCredit/Controller/Adminhtml/Index/Reimburse.php#L146-L152

If `0` is desired value please replace `null` with `0`

**Review by andrewbess — APPROVED** [review]
Hello @karyna-tsymbal-atwix 
Thank you for your contribution
Great work ✔️

**Comment by magento-engcom-team** [comment]
Hi @andrewbess, thank you for the review.
[ENGCOM-9363](https://jira.corp.magento.com/browse/ENGCOM-9363) has been created to process this Pull Request
:eight_spoked_asterisk: @andrewbess, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

File: app/code/Magento/Dhl/Model/Carrier.php [hunk 3]
```diff
@@ -939,7 +940,7 @@ protected function _getDimension($dimension, $configWeightUnit = false)
             );
         }
 
-        return round($dimension, 3);
+        return $dimension ? round($dimension, 3) : 0.0;
```
Comment (xmav) [inline app/code/Magento/Dhl/Model/Carrier.php]: type casting might be more suitable here

File: app/code/Magento/Customer/Model/Options.php [hunk 4]
```diff
@@ -47,10 +47,11 @@ public function __construct(
      */
     public function getNamePrefixOptions($store = null)
     {
-        return $this->prepareNamePrefixSuffixOptions(
-            $this->addressHelper->getConfig('prefix_options', $store),
-            $this->addressHelper->getConfig('prefix_show', $store) == NooptreqSource::VALUE_OPTIONAL
-        );
+        $options = $this->addressHelper->getConfig('prefix_options', $store);
```
Comment (xmav) [inline app/code/Magento/Customer/Model/Options.php]: please move logic to `prepareNamePrefixSuffixOptions()`

File: app/code/Magento/Dhl/Model/Carrier.php (hunk 3 above)
Comment (andrewbess) [reply app/code/Magento/Dhl/Model/Carrier.php]: Hello @xmav 
Thank you for your review
Fixed, please check.

File: app/code/Magento/Customer/Model/Options.php (hunk 4 above)
Comment (andrewbess) [reply app/code/Magento/Customer/Model/Options.php]: Hello @xmav 
Thank you for your review
Fixed, please check.

File: app/code/Magento/Customer/Model/Options.php [hunk 5]
```diff
@@ -61,10 +61,11 @@ public function getNamePrefixOptions($store = null)
      */
     public function getNameSuffixOptions($store = null)
     {
-        return $this->prepareNamePrefixSuffixOptions(
-            $this->addressHelper->getConfig('suffix_options', $store),
-            $this->addressHelper->getConfig('suffix_show', $store) == NooptreqSource::VALUE_OPTIONAL
-        );
+        $options = $this->addressHelper->getConfig('suffix_options', $store);
+        return $options !== null ? $this->prepareNamePrefixSuffixOptions(
```
Comment (xmav) [inline app/code/Magento/Customer/Model/Options.php:65]: Duplication will be removed in internal PR

**Comment by xmav** [comment]
pending delivery https://github.com/magento-commerce/magento2ce/pull/7303

**Pull request merged**